### ISSUE TITLE
NSIS crashes on Windows 7

### ISSUE THREAD

**Issue by souravm84** [electron-userland/electron-builder#2518]
* **electron-builder Version: 19.55.2**: 

* **Target Platform : x64**: 

NSIS generated by electron-builder crashes in most of the time in Windows 7. Crash Report given below.
### Problem signature:
>  Problem Event Name:	APPCRASH
  Application Name:	myapp Setup 1.0.2.exe
  Application Version:	1.0.2.0
  Application Timestamp:	584dca80
  Fault Module Name:	System.dll
  Fault Module Version:	0.0.0.0
  Fault Module Timestamp:	584dca1a
  Exception Code:	c0000005
  Exception Offset:	00001581
  OS Version:	6.1.7601.2.1.0.256.1
  Locale ID:	16393
  Additional Information 1:	0a9e
  Additional Information 2:	0a9e372d3b4ad19135b953a78882e789
  Additional Information 3:	0a9e
  Additional Information 4:	0a9e372d3b4ad19135b953a78882e789

**Comment by develar** [comment]
Is Windows 7 updated to latest SP and all system updates are installed?

Please try to set:

```
{
  "nsis": {
    "useZip": true
  }
}
```

Does it help?

**Comment by souravm84** [comment]
Thanks for your response @develar . Till now I tried adding "useZip": true in the package.json, still no luck. I have to try it on some updated windows7. 

As I said the nsis setup working properly sometimes and crashes sometimes.

**Comment by develar** [comment]
>  in the package.json,

under the key `build` and sub key `nsis`? (build.nsis.useZip = true)

**Comment by souravm84** [comment]
Yes. Please check the package.json
`"build": {
    "appId": "com.idevelopersindia.my_app",
    "publish": [
        {
          "provider": "generic",
          "url": "http://idevelopersindia.com/dist/my_app"
        }
    ],
    "win": {
      "target": [
        {
          "target": "nsis",
          "arch": [
            "x64",
            "ia32"
          ]
        }
      ]
    },
    "nsis": {
      "oneClick": false,
      "allowToChangeInstallationDirectory": true,
      "useZip": true
    }
  },`

**Comment by develar** [comment]
Does it crash if you remove `nsis` config (default installer)?

**Comment by vladimiry** [comment]
Facing similar issue https://github.com/vladimiry/protonmail-desktop-app/issues/6

**Comment by vladimiry** [comment]
Hi @develar, here is some addition information regarding issue happening cases https://github.com/vladimiry/protonmail-desktop-app/issues/6#issuecomment-360614994

**Comment by Egorikhin** [comment]
Hi @develar, could you please take a look on my PR which hopefully fix this.

**Comment by xiajingren** [comment]
hi. What is the final solution? I have a similar problem.

**Comment by jmp75** [comment]
I've had a report of a crash on  Windows 7 32 bits of an installer build with electron-builder version "^22.1.0", which should have the fix https://github.com/electron-userland/electron-builder/pull/2695 included . I set up a Win7 32 bits VM to try to reproduce the issue, but it worked impeccably even if I tried to include spaces in user names and subfolders, as this issue and related ones seem to suggest play a role.  I know Win7 32bit is past its prime, but we do have users who have logistical constraints I cannot dismiss.   I'll try to bump electron-builder to a newer version in the package.json file, and see if the installer behaves, and report if this is the case.
